### PR TITLE
21 update dfv mv data

### PR DESCRIPTION
**Pull request by bslbckr**
it's too late, I'm not quire sure whether this branch is complete. Unfortunately the changes I made are really hard to test ... Strong dependency on dfv-mv.de in this part of the code.

### CODE REVIEW

**Comment by bslbckr** [comment]
@bastrapp please have a look at my changes and merge them, if you think this is ok. Beware, I had no possibility to test my code ... I think I need  a dump of the running db during the next days, to avoid such scenarios in the future.

**Comment by bastrapp** [comment]
Thank you. There are still minor problems that should be resolved quickly. But in my opinion a rather 'big' part is missing:

If a player profile gets updated and the new profile is inactive, deleted or has no DSE (document to assure his eligibility) we need to check if this player is in any _future_ roster and remove him (and send an email to the team admins).

To avoid overhead this should be done when processing the list of players to update and should be straight forward, technically.

Defining _future_ rosters might be a little bit more tricky. All rosters of seasons with the current (or future) year should be affected. We can assume that such a "negative notification" will be issued not very often by the clubs and usually drop outs will be reported only at the "yearly reports" in january. Still, this situation could happen. It gets more complicated if the player did already play a tournament in the current season. For documentation purposes we cannot remove his name from the (past) roster but we need to make sure he is not in the _current_ roster (and even more: we need to ensure that he cannot be added to any other roster after he is being re-activated in the system).

I would suggest a date field associated with a "player to roster" mapping similar to the "date added" property that already exists. An additional field seems like a lot of overhead as it will be hardly used. Alternatively an extra table ROSTER_PLAYER_DEACTIVATION with the columns roster, player, date_deactivated could be created and used. @kaffeetrinkr what do you think?

File: backend/src/main/java/de/ultical/backend/data/DataStore.java [hunk 1]
```diff
@@ -813,4 +906,16 @@ private void modifyTeamAdmin(Team team, User admin, BiConsumer<Team, User> dbAct
             }
         }
     }
+    
+    public List<Roster> getRosterForPlayer(final DfvPlayer player) {
```
Comment (bastrapp) [inline backend/src/main/java/de/ultical/backend/data/DataStore.java:909]: @kaffeetrinkr - I don't get what this function is doing or if it was just a placeholder for something to come?!

**Comment by bslbckr** [comment]
@bastrapp thank's for the review, I forgot to complete the method and in the unit-tests either the `RosterMapper`'s method is called or the `DataStore` is mocked...

**Comment by bastrapp** [comment]
I deployed the PR ...  unfortunately there was an error when syncing... @kaffeetrinkr any ideas?

ERROR [2016-07-25 22:33:44,862] d.u.b.j.DfvDataSync:                Updating DFV profiles failed!
! java.lang.NullPointerException: null
! at de.ultical.backend.jobs.DfvProfileLoader.validateRosterParticipation(DfvProfileLoader.java:144) ~[backend-0.0.1-SNAPSHOT.jar:na]
! at de.ultical.backend.jobs.DfvProfileLoader.getDfvMvNames(DfvProfileLoader.java:74) ~[backend-0.0.1-SNAPSHOT.jar:na]
! at de.ultical.backend.jobs.DfvDataSync.doJob(DfvDataSync.java:55) ~[backend-0.0.1-SNAPSHOT.jar:na]
! at de.spinscale.dropwizard.jobs.Job.execute(Job.java:31) [backend-0.0.1-SNAPSHOT.jar:na]
! at org.quartz.core.JobRunShell.run(JobRunShell.java:202) [backend-0.0.1-SNAPSHOT.jar:na]
! at org.quartz.simpl.SimpleThreadPool$WorkerThread.run(SimpleThreadPool.java:573) [backend-0.0.1-SNAPSHOT.jar:na]

**Comment by bslbckr** [comment]
That's kind of strange. If I understand the code and the stacktrace correctly, this means that we have a roster without a team. Could you check this in the db?

Nevertheless, the JVM directly reached the code segment, which I assumed that never would be executed :smirk:

**Comment by bastrapp** [comment]
The reality can be cruel ;-)

I couldn't find a roster without a team. Everything seems in order...

**Comment by bslbckr** [comment]
I found the solution (I guess) . The getRostersForPlayer method did not retrieve the teams from the database... I fixed this in bf0b5d07c7739b0908143d43eb5ffa2ce62bc1ef

**Comment by bastrapp** [comment]
Another NPE... problem is, that we didn't get the mail even and the changes in the db still stand (I suppose). I did a backup right before the deployment of this PR, should we reapply this after fixing the issue?

ERROR [2016-07-26 07:25:57,633] d.u.b.j.DfvDataSync:                Updating DFV profiles failed!
! java.lang.NullPointerException: null
! at de.ultical.backend.app.mail.SystemMessage.getRenderedMessage(SystemMessage.java:99) ~[backend-0.0.1-SNAPSHOT.jar:na]
! at de.ultical.backend.app.MailClient.sendMail(MailClient.java:130) ~[backend-0.0.1-SNAPSHOT.jar:na]
! at de.ultical.backend.jobs.DfvProfileLoader.validateRosterParticipation(DfvProfileLoader.java:147) ~[backend-0.0.1-SNAPSHOT.jar:na]
! at de.ultical.backend.jobs.DfvProfileLoader.getDfvMvNames(DfvProfileLoader.java:74) ~[backend-0.0.1-SNAPSHOT.jar:na]
! at de.ultical.backend.jobs.DfvDataSync.doJob(DfvDataSync.java:55) ~[backend-0.0.1-SNAPSHOT.jar:na]
! at de.spinscale.dropwizard.jobs.Job.execute(Job.java:31) [backend-0.0.1-SNAPSHOT.jar:na]
! at org.quartz.core.JobRunShell.run(JobRunShell.java:202) [backend-0.0.1-SNAPSHOT.jar:na]
! at org.quartz.simpl.SimpleThreadPool$WorkerThread.run(SimpleThreadPool.java:573) [backend-0.0.1-SNAPSHOT.jar:na]
INFO  [2016-07-26 07:25:57,635] d.u.b.j.DfvDataSync:                ... Job finished!

**Comment by bastrapp** [comment]
Guess I fixed it in a43bcf9b7304a852a1e5eba1d806cf5823f33cee and promptly an email was send... good. So the changes didn't go through because the email was not send? Do you agree? Or should we just do the db restore?

**Pull request merged**